boltdesignsystem/bolt
# Migrate Accordion web component to now render using LitElement

## Jira
http://vjira2:8080/browse/BDS-2125

## Summary
Migrates our existing Accordion web component to now render using LitElement. Addresses catastrophic SSR hydration issues that can sometimes occur when asynchronously loading the component's JS / being lazily rendered (for example, [as seen here](https://boltdesignsystem-pubfgtblg.now.sh/pattern-lab/patterns/02-components-accordion-40-accordion-content-variations/02-components-accordion-40-accordion-content-variations.html)).

## Details
As far as the Accordion component goes, this is more or less a 1:1 upgrade with only minor logic tweaks required, plus a new interaction part of the existing Jest test added as well.

Note: As an added bonus, updating the `spacing` prop (when an accordion item is already open -- ex. going from small to large spacing, or vice-versa) now correctly recalculates the height of the open panel which doesn't happen in the current version...

## How to test
- Confirm that with these upgrades, the Accordion component renders correctly (despite it now being lazily loaded on this branch + renders using `<ssr-keep>`.

- Also confirm that contextual props (ex. `single`, `spacing`, etc) also trigger the Accordion Items to re-render as expected.
